src/headers/languages/metadata.rs

```diff
@@ -0,0 +1,470 @@
+#![cfg_attr(rustfmt, rustfmt::skip)]
+
+use super::*;
+use crate::layout;
+use crate::utils::DisplayFromFn;
+use core::fmt::Display;
+use std::io::Write;
+
+pub struct Metadata;
+
+impl<F: Fn(&mut dyn Write) -> io::Result<()>> DisplayFromFn<F> {
+    fn indented_lines(&self) -> String {
+        self.to_string().lines().map(|line| format!("    {line}\n")).collect()
+    }
+}
+
+impl HeaderLanguage for Metadata {
+    fn declare_simple_enum(
+        self: &'_ Self,
+        this: &dyn HeaderLanguage,
+        ctx: &'_ mut dyn Definer,
+        docs: Docs<'_>,
+        self_ty: &'_ dyn PhantomCType,
+        backing_integer: Option<&dyn PhantomCType>,
+        variants: &'_ [EnumVariant<'_>],
+    ) -> io::Result<()> {
+        let ref indent = Indentation::new(4);
+        mk_out!(indent, ctx.out());
+
+        if let _ = indent.scope() {
+            out!((",{{"));
+
+            if let _ = indent.scope() {
+                out!(("\"kind\": \"Enum\","));
+
+                self.emit_docs(ctx, docs, indent)?;
+
```

Comment:
nit: I picked a random line but the whitespacing in this file is kinda weird. Lots of newlines without any discernible consistent pattern for when to have a newline or not, at least to my eyes.